src/resolvers.js

```diff
@@ -276,7 +277,10 @@ export class NewInstance {
   * @return Returns the matching instance from the parent container
   */
   get(container) {
-    const instance = container.invoke(this.key);
+    let dynamicDependencies = this.dynamicDependencies.length > 0 ?
+      this.dynamicDependencies.map(dependency => dependency["protocol:aurelia:resolver"] ?
+        dependency.get(container) : container.invoke(dependency)) : undefined;
```

Comment:
If you can make that change, we can probably get this in. Ping me when you've updated the PR please. Thank you!